vm-mock/vm.js

```diff
@@ -774,6 +775,12 @@ export default function createMockedABI(
         const hash = sha3.keccak256.arrayBuffer(data);
         return newArrayBuffer(hash);
       },
+      assembly_script_set_chain_id(value) {
+        chainIdMock = value;
+      },
+      assembly_script_chain_id() {
+        return chainIdMock;
+      },
```

Comment:
I don't see unit test for this, are we sure it works? most of functions here handle params and return values as pointers